test/core/ext/filters/event_engine_client_channel_resolver/resolver_fuzzer.cc

```diff
@@ -154,40 +154,45 @@ class FuzzingResolverEventEngine
     void LookupHostname(LookupHostnameCallback on_resolve,
                         absl::string_view /* name */,
                         absl::string_view /* default_port */) override {
-      CheckAndSetOrphan(ExecutionStep::DURING_LOOKUP_HOSTNAME);
+      CheckAndSetOrphan(ExecutionStep::DURING_LOOKUP_HOSTNAME, engine_);
       if (!engine_->has_been_orphaned_) {
-        engine_->runner_.Run([this, cb = std::move(on_resolve)]() mutable {
-          CheckAndSetOrphan(ExecutionStep::AFTER_LOOKUP_HOSTNAME_CALLBACK);
-          cb(engine_->hostname_responses_);
-        });
+        engine_->runner_.Run(
+            [engine = engine_, cb = std::move(on_resolve)]() mutable {
+              CheckAndSetOrphan(ExecutionStep::AFTER_LOOKUP_HOSTNAME_CALLBACK,
+                                engine);
+              cb(engine->hostname_responses_);
+            });
       }
     }
     void LookupSRV(LookupSRVCallback on_resolve,
                    absl::string_view /* name */) override {
-      CheckAndSetOrphan(ExecutionStep::DURING_LOOKUP_SRV);
+      CheckAndSetOrphan(ExecutionStep::DURING_LOOKUP_SRV, engine_);
       if (!engine_->has_been_orphaned_) {
-        engine_->runner_.Run([this, cb = std::move(on_resolve)]() mutable {
-          CheckAndSetOrphan(ExecutionStep::AFTER_LOOKUP_SRV_CALLBACK);
-          cb(engine_->srv_responses_);
+        engine_->runner_.Run([engine = engine_,
+                              cb = std::move(on_resolve)]() mutable {
+          CheckAndSetOrphan(ExecutionStep::AFTER_LOOKUP_SRV_CALLBACK, engine);
+          cb(engine->srv_responses_);
         });
       }
     }
     void LookupTXT(LookupTXTCallback on_resolve,
                    absl::string_view /* name */) override {
-      CheckAndSetOrphan(ExecutionStep::DURING_LOOKUP_TXT);
+      CheckAndSetOrphan(ExecutionStep::DURING_LOOKUP_TXT, engine_);
       if (!engine_->has_been_orphaned_) {
-        engine_->runner_.Run([this, cb = std::move(on_resolve)]() mutable {
-          CheckAndSetOrphan(ExecutionStep::AFTER_LOOKUP_TXT_CALLBACK);
-          cb(engine_->txt_responses_);
+        engine_->runner_.Run([engine = engine_,
+                              cb = std::move(on_resolve)]() mutable {
+          CheckAndSetOrphan(ExecutionStep::AFTER_LOOKUP_TXT_CALLBACK, engine);
+          cb(engine->txt_responses_);
         });
       }
     }
 
    private:
-    void CheckAndSetOrphan(ExecutionStep current_execution_step) {
-      if (engine_->should_orphan_at_step_ == current_execution_step) {
-        *engine_->done_resolving_ = true;
-        engine_->has_been_orphaned_ = true;
+    static void CheckAndSetOrphan(ExecutionStep current_execution_step,
```

Comment:
Since this doesn't touch any `FuzzingDNSResolver` internals, and since it might be called after the `FuzzingDNSResolver` object is destroyed, I suggest moving this method into `FuzzingResolverEventEngine`. The signature can remain the as it was before.

Reply:
Good point! Done.